runtime/src/main/java/com/dylibso/chicory/runtime/HostFunction.java

```diff
@@ -3,50 +3,16 @@
 import com.dylibso.chicory.wasm.types.ValueType;
 import java.util.List;
 
-public class HostFunction implements FromHost {
-    private final WasmFunctionHandle handle;
-    private final String moduleName;
-    private final String fieldName;
-    private final List<ValueType> paramTypes;
-    private final List<ValueType> returnTypes;
-
+/**
+ * A HostFunction is an ExternalFunction that has been defined by the host.
```

Comment:
If we keep the distinction, we should have also, at least `HostMemory` too.

In general, I would go for the simplest solution and just use `ExternalFunction` everywhere.

Reply:
I feel like it is worth keeping the distinction. 

IMO the two might diverge at some point i.e. while we want HostFunction to be user-facing, the internals/mechanics of a generic ExternalFunction (e.g. when it deals with exports from other instances) might be different. This is not necessarily (only) a matter of implementation, but also a matter of clarity.

OTOH I don't think an ExternalMemory defined by the Host is different from any other host memory.

Again, I'd rather keep the concept of HostFunction because it's what people are familiar with; but maybe this can be solved with a named factory somewhere (as mentioned in the body of the PR) :)

Reply:
for a more authoritative source, notice [how HostFunctions are separate from FunctionInstances in the spec](https://webassembly.github.io/spec/core/exec/runtime.html#function-instances), while there is no distinction between host-supplied and "plain" [table instances](https://webassembly.github.io/spec/core/exec/runtime.html#table-instances), [memory instances](https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances) or [global instances](https://webassembly.github.io/spec/core/exec/runtime.html#global-instances).

Reply:
I'm convinced, thanks for explaining and finding the sources! Appreciated :+1: